conans/client/tools/scm.py

```diff
@@ -62,14 +62,38 @@ def run(self, command):
                         return self._runner(command)
 
     def get_url_with_credentials(self, url):
-        if not self._username or not self._password:
+        if not self._username and not self._password:
             return url
-        if urlparse(url).password:
+        url_parsed = urlparse(url)
+        if url_parsed.scheme == "ssh" and (url_parsed.password or self._password):
```

Comment:
I think` url_parsed.scheme` does not work properly for scp-like syntax like: `git@github.com:memsharded/hello.git`

Reply:
I don't know how to detect this if the scheme is not provided

Reply:
Maybe we could detect the syntax with a regular expression, converting to a ssh:// url and then passing to urlparse? WDYT?

Reply:
Maybe this could be enough if the scheme is "":

```python
if "@" in url and ":" in url:
    return "ssh://" + url.replace(":", "/")
```

Reply:
I did some tests and it is very fragile with that conditional. I t might work with a regex but I doesn't feel good manipulating the URL that much. It will be weird for the user to change the provided URL. I will need a second opinion about it @memsharded

Reply:
Yes, something like [this](https://github.com/golang/go/blob/979e1376096a4c1c7525f33dd4e76414f63c93fb/src/cmd/go/internal/vcs/vcs.go#L206) would be more robust in case we want to do this conversion.

Reply:
This whole feature feels a bit weird to me. Like, for a user defined the URL they want is a one liner, lets say:

```python
url = "https://{user}:{passwd}@someurl.com/repo".format(user=os.getenv("USER"), passwd=os.getenv("PASSWD")
```
Am I missing something?
While on our side is complicated and fragile, trying to cover all cases. Maybe we want to re-think the whole UX.

Reply:
There are two sources for this URL: user-input or `auto` keyword. Parsing all formats can be very challenging ([😮  there are a lot](https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats)). Maybe we should:
 * check it matches some regex for well-known formats (those in Github)
 * if it doesn't match, the URL will be used without any transformation. If user/password are provided, print a warning telling the user that _"URL format unknown, user and password won't be used to clone"_ (and in this case, do not store user/pasword?).